us_imputation_benchmarking/models/imputer.py

```diff
@@ -0,0 +1,88 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Set, Union
+
+import numpy as np
+import pandas as pd
+
+from us_imputation_benchmarking.utils.logging_utils import get_logger
+
+
+class Imputer(ABC):
+    """
+    Abstract base class for imputation models.
+
+    All imputation models should inherit from this class and implement
+    the required methods.
+    """
+
+    def __init__(self, random_seed=None) -> None:
+        """Initialize the imputer model."""
+        self.predictors: Optional[List[str]] = None
+        self.imputed_variables: Optional[List[str]] = None
+        self.seed = random_seed
+        self.logger = get_logger(
+            f"{self.__class__.__module__}.{self.__class__.__name__}"
+        )
+
+    def _validate_data(
+        self, data: pd.DataFrame, columns: List[str], context: str
+    ) -> None:
+        """Validate that all required columns are in the data.
+
+        Args:
+            data: DataFrame to validate
+            columns: Column names that should be present
+            context: Context string for error messages (e.g., 'training', 'prediction')
```

Comment:
I don't think we need this context concept? Will be obvious from the stack trace which bit failed.